pkg/download/download.go

```diff
@@ -63,8 +95,10 @@ func (p *protocol) Info(ctx context.Context, mod, ver string) ([]byte, error) {
 	return info, nil
 }
 
-func (p *protocol) fillCache(ctx context.Context, mod, ver string) error {
+func (p *protocol) fillCache(mod, ver string) error {
 	const op errors.Op = "protocol.fillCache"
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*10)
```

Comment:
@michalpristas i purposefully removed that, because you want the fillCache to have its own context. When you are downloading from upstream, or uploading to storage, you don't want to cancel that if the user abandoned the request.

Reply:
oh you mean the job struct and not the old ctx? Yeah I can do that, but what's the gain?

Reply:
so you can pass it down to the job and do proper tracing/cancellation. instead of using ctx.Background

Reply:
the cancellation - not propagating req.cancel is a valid point I too thought about that. BUt couldn't figure out which way is the correct one

Reply:
@michalpristas I'd like to see the cacheFIll as part of the whole tracing, but im not sure if OpenCensus ignores a cancelled context? Or maybe it can copy a context value to a new one? In any case, OpenCensus is not yet part of our codebase so is it cool if I make an issue about this? 

As for cancellation, what would be a proper cancellation mechanism besides an _independent_ time out of 10 minutes?

Reply:
i was thinking about canceling with the request. 
let's keep it as you have it now. in case of huge repos, it would timeout with every request leading to impossible cache fill.

Reply:
@michalpristas sounds good. Here's the tracing issue https://github.com/gomods/athens/issues/526
THANK YOU